src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/AppleGetDeviceCommand.cs

```diff
@@ -0,0 +1,116 @@
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Apple;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.CLI.CommandArguments;
+using Microsoft.DotNet.XHarness.Common.CLI.Commands;
+using Microsoft.DotNet.XHarness.Common.Logging;
+using Microsoft.DotNet.XHarness.iOS.Shared;
+using Microsoft.DotNet.XHarness.iOS.Shared.Execution;
+using Microsoft.DotNet.XHarness.iOS.Shared.Hardware;
+using Microsoft.DotNet.XHarness.iOS.Shared.Utilities;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Apple
+{
+    internal class AppleGetDeviceCommand : XHarnessCommand
+    {
+        private readonly AppleGetDeviceArguments _arguments = new();
+
+        protected override XHarnessCommandArguments Arguments => _arguments;
+
+        protected override string CommandUsage { get; } = "apple device [TARGET] [OPTIONS]";
+
+        private const string CommandHelp = "Finds the UDID of a device/simulator for given target";
+        protected override string CommandDescription { get; } = @$"
+{CommandHelp}
+ 
+Arguments:
+";
+
+        public AppleGetDeviceCommand() : base("device", true, CommandHelp)
+        {
+        }
+
+        protected override async Task<ExitCode> InvokeInternal(ILogger logger)
+        {
+            var processManager = new MlaunchProcessManager(_arguments.XcodeRoot, _arguments.MlaunchPath);
+
+            var log = new CallbackLog(m => logger.LogDebug(m));
+            TestTargetOs target;
+
+            try
+            {
+                target = ParseTarget();
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e.Message);
+                return ExitCode.INVALID_ARGUMENTS;
+            }
+
+            try
+            {
+                if (target.Platform.IsSimulator())
+                {
+                    var simulatorLoader = new SimulatorLoader(processManager);
+
+                    // TODO: Handle companion mode
+                    var (simulator, _) = await simulatorLoader.FindSimulators(target, log, 3, true);
+                    Console.WriteLine(simulator.UDID);
+                }
+                else
+                {
+                    var hardwareDeviceLoader = new HardwareDeviceLoader(processManager);
+                    var device = await hardwareDeviceLoader.FindDevice(target.Platform.ToRunMode(), log, false);
+                    Console.WriteLine(device.UDID);
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e.ToString());
+                return ExitCode.DEVICE_NOT_FOUND;
+            }
+
+            return ExitCode.SUCCESS;
+        }
+
+        private TestTargetOs ParseTarget()
+        {
+            if (ExtraArguments.Count() != 1)
```

Comment:
probably `<1` would be better? or you want to validate that there is no more other arguments?

Reply:
Yeah, I only want 1. I will improve the message